machine_learning/loss_functions/categorical_cross_entropy.py

```diff
@@ -0,0 +1,78 @@
+"""
+Categorical Cross-Entropy Loss
+
+This function calculates the Categorical Cross-Entropy Loss between true class
+labels and predicted class probabilities.
+
+Formula:
+Categorical Cross-Entropy Loss = -Σ(y_true * log(y_pred))
```

Comment:
```suggestion
Categorical Cross-Entropy Loss = -Σ(y_true * ln(y_pred))
```
Let's use `ln` instead of `log` here in order to make it unambiguous that we're using the natural log